### PR TITLE
update hokusai-required-version check to use PEP-440 version specifiers

### PR DESCRIPTION
**Pull request by izakp**
Also add the [packaging library](https://packaging.pypa.io/en/latest/) that provides the canonical implementation of PEP-440 for use in setuptools / pip, etc...

I think this is a lot more robust, flexible and clearer as the [versioning scheme](https://www.python.org/dev/peps/pep-0440/#version-scheme) already used by Hokusai as well as the [behavior of version specifiers](https://www.python.org/dev/peps/pep-0440/#version-specifiers), are well-specified and documented in PEP-440.

I was thinking about rebasing once this was merged to `master` but since it will only create a release tag for v0.5.1 on merge `master` -> `release` I think it's okay to go once we sanity-check the PyInstaller build on S3

### CODE REVIEW

**Comment by joeyAghion** [comment]
Fancy!

The required version will apply to _all_ commands, right?

**Comment by izakp** [comment]
> The required version will apply to all commands, right?

All except those that opt-out by invoking the command decorator `@command(config_check=False)`, so presently only `hokusai setup` and `hokusai version` are exempt

File: test/unit/test_config.py [hunk 1]
```diff
@@ -39,9 +39,32 @@ def test_create(self):
 
   def test_check_hokusai_required_version(self):
     self.assertTrue(self.config._check_required_version(None, '0.0.1'))
-    self.assertTrue(self.config._check_required_version('0.0.1', '0.0.1'))
-    self.assertTrue(self.config._check_required_version('0.0.1', '0.0.2'))
-    self.assertFalse(self.config._check_required_version('1.0', '0.2'))
+    with self.assertRaises(HokusaiError):
+      self.config._check_required_version('0.0.1', '0.0.1')
```
[the remark this answers is not in the document]
Comment (izakp) [reply test/unit/test_config.py]: Yes as `0.0.1` is not a valid version specifier according to PEP-440, you'd need to use something like `==0.0.1` or `>=0.0.1`

**Comment by izakp** [comment]
@dleve123 

> Is this technically a breaking change compared to the implementation in #127 right? Are you just gonna include this in the 0.5.1 release?

Yes - the implementation in #127 is only in master and 0.5.1 hasn't been released yet so this basically overrides the other implementation.

I was thinking about rebasing master so the commit fbf743f7157b35dcb590fc491326fc4f3e5c3afa "prepare version 0.5.1" comes after these commits, for clarity about what is in 0.5.1, but it's not strictly necessary as the tag `v0.5.1` will be created on the merge to release.  I should however, update the CHANGELOG.

**Comment by dleve123** [comment]
> I was thinking about rebasing master so the commit fbf743f "prepare version 0.5.1" comes after these commits, for clarity about what is in 0.5.1, but it's not strictly necessary as the tag v0.5.1 will be created on the merge to release. I should however, update the CHANGELOG.

I wouldn't recommend rebasing `master` as it seems pretty atypical (and might break tooling?), and I think it would be good for the Git history to tell the story of this change. Generally including this in 0.5.1 sounds good though!

Merge at your discretion.

**Pull request merged**